File: services/gestion-svc/src/main/java/edu/utn/frsf/isi/dan/gestion/service/TarifaService.java
package edu.utn.frsf.isi.dan.gestion.service;

import edu.utn.frsf.isi.dan.gestion.dto.TarifaDTORequest;
import edu.utn.frsf.isi.dan.gestion.dto.TarifaDTOResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TarifaService {

    TarifaDTOResponse crearTarifa(TarifaDTORequest request);

    TarifaDTOResponse buscarTarifaPorId(Integer id);

    Page<TarifaDTOResponse> buscarTarifas(Pageable pageable);

    void eliminarTarifa(Integer id);
}

File: frontend/src/services/tarifa.service.ts
import api from "./api";
import type { Tarifa, TarifaCreateRequest, PageResponse } from "@/types/hotel";

const BASE = "/gestion/tarifas";

export const tarifaService = {
  buscarTarifas: (params: { page?: number; size?: number }) =>
    api.get<PageResponse<Tarifa>>(BASE, { params }).then((r) => r.data),

  buscarTarifaPorId: (id: number) =>
    api.get<Tarifa>(`${BASE}/${id}`).then((r) => r.data),

  crearTarifa: (data: TarifaCreateRequest) =>
    api.post<Tarifa>(BASE, data).then((r) => r.data),

  eliminarTarifa: (id: number) =>
    api.delete(`${BASE}/${id}`),
};
